src/java.base/share/classes/java/lang/invoke/MethodHandleProxies.java

```diff
@@ -292,7 +292,7 @@ private static boolean isObjectMethod(Method m) {
     private static Object callObjectMethod(Object self, Method m, Object[] args) {
         assert(isObjectMethod(m)) : m;
         return switch (m.getName()) {
-            case "toString" -> self.getClass().getName() + "@" + Integer.toHexString(self.hashCode());
+            case "toString" -> java.util.Objects.toDefaultString(self);
```

Comment:
It might be better if toString is changed to invoke toIdentityString(self), only because hashCode returns the identity hash code, it doesn't invoke hashCode().

Reply:
Yes; that looks like an inconsistency/bug. Updated in a subsequent push.

Reply:
Yes, it's a bug and it should call the one with identity hash code.   The `toString` method of a Proxy instance should return the result of the default implementation of `Object::toString` method as specified.